arangod/Graph/GraphOperations.cpp

```diff
@@ -844,8 +856,74 @@ std::pair<OperationResult, bool> GraphOperations::validateEdgeContent(
   VPackSlice toStringSlice = document.get(StaticStrings::ToString);
   OperationOptions options(ExecContext::current());
 
+  // Validate from & to slices or
+  // validate from || to slices
+  auto validateSlices =
+      [&](std::optional<VPackSlice> fromSlice,
+          std::optional<VPackSlice> toSlice) -> OperationResult {
+    if (fromSlice.has_value()) {
+      std::string fromString = fromSlice.value().copyString();
+
+      // _from part
+      size_t pos = fromString.find('/');
+      if (pos != std::string::npos) {
+        fromCollectionName = fromString.substr(0, pos);
+        // check if vertex collections are part of the graph definition
+        auto foundFromRes = checkVertexCollectionAvailability(
+            fromCollectionName, VertexValidationOrigin::FROM_ATTRIBUTE);
+        if (foundFromRes.fail()) {
+          return OperationResult(foundFromRes, options);
+        }
+
+        fromCollectionKey = fromString.substr(pos + 1, fromString.length());
```

Comment:
nit: while this works, the usage of std::strings here is not efficient, as we create up to 3 std::strings for the _from and up to 3 std::strings for the _to parts. Using std::string_views here would have done it as well, but it may require the validation function `checkVertexCollectionAvailability` to use std::string_view as well.
So I am not insisting on the change now.

Reply:
I do agree, but yes this will be a bigger change. Quickly tested it and a lot of things need to be adjusted then. So let's do this in a follow up pr.